### ISSUE TITLE
axe-core triggers `elementFromPoint` SupportError for `<dialog>` elements in jsdom environment

### ISSUE THREAD

**Issue by scurker** [dequelabs/axe-core#4997]
### Product

axe-core

### Product Version

4.10.2

### Latest Version

- [x] I have tested the issue with the latest version of the product

#### Expectation

When running axe-core in a jsdom-based test environment, using a `<dialog>` element should behave consistently with other container elements such as `<div>`.

#### Actual

Using the same test structure with a `<div>` instead of `<dialog>` produces a different outcome.

When `<dialog>` is used, certain rules are skipped due to a SupportError, which can lead to false negatives in accessibility tests. The scan completes successfully, but applicable rules are not evaluated.

Example stack trace:

```
{
  id: 'button-name',
  impact: null,
  description: 'Ensure buttons have discernible text',
  message: '`TypeError: document.elementFromPoint is not a function` - feature unsupported in your environment. Skipping button-name rule.',
  stack: `error: SupportError {
    name: 'SupportError',
    cause: TypeError: document.elementFromPoint is not a function
        at Document.elementsFromPoint (/projects/code/node_modules/jest-axe/node_modules//projects/code/axe.js:19286:36)
        at /projects/code/node_modules/jest-axe/node_modules//projects/code/axe.js:10002:30
        at Array.filter (<anonymous>)
        at getNodesMatchingExpression (/projects/code/node_modules/jest-axe/node_modules//projects/code/axe.js:18198:37)
        at querySelectorAllFilter (/projects/code/node_modules/jest-axe/node_modules//projects/code/axe.js:19377:19)
        at getModalDialogMemoized (/projects/code/node_modules/jest-axe/node_modules//projects/code/axe.js:10000:21)
        at memoized (/projects/code/node_modules/jest-axe/node_modules//projects/code/axe.js:1646:29)
        at isInertSelfMemoized (/projects/code/node_modules/jest-axe/node_modules//projects/code/axe.js:10070:27)
        at /projects/code/node_modules/jest-axe/node_modules//projects/code/axe.js:1617:30
        at _isInert (/projects/code/node_modules/jest-axe/node_modules//projects/code/axe.js:10061:16)
        at isVisibleToScreenReadersMemoized (/projects/code/node_modules/jest-axe/node_modules//projects/code/axe.js:13049:32)
        at /projects/code/node_modules/jest-axe/node_modules//projects/code/axe.js:1615:29
        at _isVisibleToScreenReaders (/projects/code/node_modules/jest-axe/node_modules//projects/code/axe.js:13046:14)
        at /projects/code/node_modules/jest-axe/node_modules//projects/code/axe.js:27733:18
        at Array.filter (<anonymous>)
        at Rule.gather (/projects/code/node_modules/jest-axe/node_modules//projects/code/axe.js:27732:29)
        at Rule.gatherAndMatchNodes (/projects/code/node_modules/jest-axe/node_modules//projects/code/axe.js:27939:24)
        at Rule.run2 [as run] (/projects/code/node_modules/jest-axe/node_modules//projects/code/axe.js:27794:22)
        at /projects/code/node_modules/jest-axe/node_modules//projects/code/axe.js:28570:14
        at pop (/projects/code/node_modules/jest-axe/node_modules//projects/code/axe.js:8169:18)
        at Object.defer (/projects/code/node_modules/jest-axe/node_modules//projects/code/axe.js:8191:11)
        at /projects/code/node_modules/jest-axe/node_modules//projects/code/axe.js:28284:27
        at Array.forEach (<anonymous>)
        at Audit.run (/projects/code/node_modules/jest-axe/node_modules//projects/code/axe.js:28283:23)
        at /projects/code/node_modules/jest-axe/node_modules//projects/code/axe.js:28652:15
        at pop (/projects/code/node_modules/jest-axe/node_modules//projects/code/axe.js:8169:18)
        at Object.defer (/projects/code/node_modules/jest-axe/node_modules//projects/code/axe.js:8191:11)
        at Object.runRules [as _runRules] (/projects/code/node_modules/jest-axe/node_modules//projects/code/axe.js:28651:9)
        at Object.run4 [as run] (/projects/code/node_modules/jest-axe/node_modules//projects/code/axe.js:28902:11)
        at /projects/code/node_modules/jest-axe/index.js:79:15
        at new Promise (<anonymous>)
        at axe (/projects/code/node_modules/jest-axe/index.js:78:12)
        at /projects/code/packages/mezzanine/components/dialog/__tests__/Modal-tests.tsx:122:50
        at Generator.next (<anonymous>)
        at /projects/code/packages/mezzanine/components/dialog/__tests__/Modal-tests.tsx:8:71
        at new Promise (<anonymous>)
        at Object.<anonymous>.__awaiter (/projects/code/packages/mezzanine/components/dialog/__tests__/Modal-tests.tsx:4:12)
        at Object.<anonymous> (/projects/code/packages/mezzanine/components/dialog/__tests__/Modal-tests.tsx:120:60)
        at Promise.then.completed (/projects/code/node_modules/@jest/core/node_modules/jest-circus/build/utils.js:298:28)
        at new Promise (<anonymous>)
        at callAsyncCircusFn (/projects/code/node_modules/@jest/core/node_modules/jest-circus/build/utils.js:231:10)
        at _callCircusTest (/projects/code/node_modules/@jest/core/node_modules/jest-circus/build/run.js:316:40)
        at _runTest (/projects/code/node_modules/@jest/core/node_modules/jest-circus/build/run.js:252:3)
        at _runTestsForDescribeBlock (/projects/code/node_modules/@jest/core/node_modules/jest-circus/build/run.js:126:9)
        at _runTestsForDescribeBlock (/projects/code/node_modules/@jest/core/node_modules/jest-circus/build/run.js:121:9)
        at run (/projects/code/node_modules/@jest/core/node_modules/jest-circus/build/run.js:71:3)
        at runAndTransformResultsToJestFormat (/projects/code/node_modules/@jest/core/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapterInit.js:122:21)
        at jestAdapter (/projects/code/node_modules/@jest/core/node_modules/jest-circus/build/legacy-code-todo-rewrite/jestAdapter.js:79:19)
        at runTestInternal (/projects/code/node_modules/jest-runner/build/runTest.js:367:16)
        at runTest (/projects/code/node_modules/jest-runner/build/runTest.js:444:34)`
}
```

The following test passes with multiple incomplete issues due to the above error, even though it contains a button without a discernible name and would be expected to fail:

```tsx
test("should have no a11y violations when dialog is open", async () => {
  render(
    <dialog open data-testid="dialog">
      <button></button>
    </dialog>
  );

  const dialog = screen.getByTestId("dialog");
  const results = await axe(dialog);

  // Passes due to the error raised
  expect(results).toHaveNoViolations();
});
```

By contrast, the equivalent test using a `<div>` behaves as expected and fails correctly:

```tsx
test("should have no a11y violations for div", async () => {
  render(
    <div data-testid="div">
      <button></button>
    </div>
  );

  const element = screen.getByTestId("div");
  const results = await axe(element);

  // This test fails correctly
  expect(results).toHaveNoViolations();
});
```

#### How to Reproduce

A minimal, runnable reproduction is available here:
https://codesandbox.io/p/sandbox/hardcore-cohen-fmy6cz

#### Additional context

I acknowledge that this may not strictly be an axe-core issue and may stem from limitations in jsdom. There is an existing jsdom issue related to [`elementFromPoint()`](https://github.com/jsdom/jsdom/issues/1435). 

This issue has also [been discussed in jest-axe](https://github.com/NickColley/jest-axe/issues/290) and the maintainer has said they [won't fix](https://github.com/NickColley/jest-axe/issues/290#issuecomment-2892331270).

The reason I am flagging this is the risk of false negatives. In the provided reproduction, the `<div>` example fails accessibility checks correctly, while the `<dialog>` example passes due to skipped rules. If I were not explicitly checking for incomplete or skipped rules in the axe output, I would reasonably assume that the `<dialog>` implementation is accessible.

This makes it easy for accessibility regressions to go unnoticed in jsdom-based test environments when `<dialog>` is used, even though equivalent markup in other container elements would fail.

I understand this may stem from jsdom limitations rather than an axe-core defect. However, the element-specific behavior and silent pass outcome make this case particularly easy to miss for consumers, which is why I wanted to surface it for discussion.
